### PR TITLE
[civ2][flakiness/7] move core flaky tests to civ2

### PR DESCRIPTION
**Pull request by can-anyscale**
Move core flaky tests in civ01 to run in a single step in civ02. Mark that job as soft fail for now.

Test:
- CI

### CODE REVIEW

File: .buildkite/pipeline.build_core.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 #ci:group=:book: core tests (civ2)
 
+- label: ":python: small & medium"
```
Comment (aslonnie) [inline .buildkite/pipeline.build_core.yml:3]: this label here also feels weird, probably needs more rebase/merge audit?

File: ci/ray_ci/runner.py [hunk 2]
```diff
@@ -72,9 +87,6 @@ def _get_test_targets(
     """
     Get test targets to run for a particular shard
     """
-    if not yaml_dir:
-        yaml_dir = os.path.join(bazel_workspace_dir, "ci/ray_ci")
-
```
Comment (aslonnie) [inline ci/ray_ci/runner.py:77]: why are these removed? I thought you say this is the right place?

now it seems that it will be executed for multiple times?

File: .buildkite/pipeline.build_core.yml (hunk 1 above)
Comment (can-anyscale) [reply .buildkite/pipeline.build_core.yml]: fixed now

File: ci/ray_ci/runner.py [hunk 3]
```diff
@@ -32,14 +32,29 @@
...
+    size: str,
+    run_flaky_tests: bool,
 ) -> None:
     if not bazel_workspace_dir:
         raise Exception("Please use `bazelisk run //ci/ray_ci`")
     os.chdir(bazel_workspace_dir)
 
-    test_targets = _get_test_targets(targets, team, concurrency, shard, size)
+    if run_flaky_tests:
+        test_targets = _get_flaky_test_targets(team)
```
Comment (can-anyscale) [inline ci/ray_ci/runner.py:55]: [1]

File: ci/ray_ci/runner.py (hunk 2 above)
Comment (aslonnie) [reply ci/ray_ci/runner.py:77]: then maybe remove the `Optional` and default `None` value in the arg list.

**Review by aslonnie — APPROVED** [review]
